core/subsurface-qt/diveobjecthelper.cpp

```diff
@@ -249,6 +249,28 @@ QStringList getFullCylinderList()
 	return cylinders;
 }
 
+QString formatSalinity(const dive *d)
+{
+	int salinity = get_dive_salinity(d);
+	if (!salinity)
+		return QString();
+        return QString("%1g/ℓ").arg(salinity / 10.0);
+}
+
+QString formatWaterType(const dive *d)
+{
+	int salinity = get_dive_salinity(d);
+	if (salinity < 10050)
+		return waterTypes[FRESHWATER];
+	else if (salinity < 10190)
+		return waterTypes[SALTYWATER];
+	else if (salinity < 10210)
+		return waterTypes[EN13319WATER];
+	else
+		// Should we return something like "Super Salty" here?
```

Comment:
It seems like we could use ``formatWaterType`` in ``desktop-widgets/tab-widgets/TabDiveInformation.cpp`` as well - except I think each places wants a different thing for the else case. Perhaps with a second param which is "What to return in the else case"? Or maybe 2 different case statements aren't super terrible.

Reply:
why not return `waterTypes[SALTWATER]` here?

Reply:
@neolit123 I think because the numbers in the if statements have saltwater sandwiched between fresh and en13319 (which seems odd to me - so returning salt for values between 10050 and 10189, then en13319 from 10190 to 10219 then saltwater again above that seems ... strange? Maybe this is just indicating that this mapping is less than ideal anyway?

I don't have strong opinions here and happy to do whatever - was mostly just carrying over existing logic.

Reply:
Note that one is saltYwater, the other is saltwater. Whether that makes sense, is another matter.

Reply:
i think SALTWATER makes sense as the "saltiest" value:
- fresh
- salty
- EN13319
- salt

Reply:
So this "progression" was introduced by @willemferguson a while ago. I felt it was a bit excessive, but he felt strongly about it - and since I don't really use this at all, I didn't care (and no one else spoke up)...

Reply:
Ah- looking at ``TabDiveInformation::on_waterTypeCombo_activated`` - salt and salty were used for the two different values on either side of ``EN13319``. 

Shearwater's FAQ says 
```
    Fresh Water = 1000kg/m³
    EN13319 = 1020 kg/m³
    Salt Water = 1030 kg/m³
```

which seems in keeping with what we're doing here somewhat. I think keeping this as-is for now and then doing a followup change to rationalize the two and potentially put in a comment as to why we're picking what we're picking would be nice. I'll work on that.

Reply:
Nod. I agree that ``Saltwater`` makes the most sense for the value above ``EN13319``. Perhaps we should rename "SaltY" to "brackish" which is the correct term for "not fresh but not salt". I think that needs to also update the code that does use-settable salinity.

Reply:
i don't know what `brackish` means.

i think we should just replace:
`// Should we return something like "Super Salty" here?`

with `return waterTypes[SALTWATER];` for now.

Reply:
Yeah - done. I'll do the salty vs brackish thing as an alternate PR so it can be discussed appropriately.